model/cluster_utility.go

```diff
@@ -37,120 +35,59 @@ const (
 	GitlabOAuthTokenKey = "GITLAB_OAUTH_TOKEN"
 )
 
-var (
+// DefaultUtilityVersions holds the default values for all of the HelmUtilityVersions
+var DefaultUtilityVersions map[string]*HelmUtilityVersion = map[string]*HelmUtilityVersion{
 	// PrometheusOperatorDefaultVersion defines the default version for the Helm chart
-	PrometheusOperatorDefaultVersion = &HelmUtilityVersion{Chart: "9.4.4", ValuesPath: ""}
+	PrometheusOperatorCanonicalName: {Chart: "9.4.4", ValuesPath: ""},
 	// ThanosDefaultVersion defines the default version for the Helm chart
-	ThanosDefaultVersion = &HelmUtilityVersion{Chart: "3.2.2", ValuesPath: ""}
+	ThanosCanonicalName: {Chart: "3.2.2", ValuesPath: ""},
 	// NginxDefaultVersion defines the default version for the Helm chart
-	NginxDefaultVersion = &HelmUtilityVersion{Chart: "2.15.0", ValuesPath: ""}
+	NginxCanonicalName: {Chart: "2.15.0", ValuesPath: ""},
 	// NginxInternalDefaultVersion defines the default version for the Helm chart
-	NginxInternalDefaultVersion = &HelmUtilityVersion{Chart: "2.15.0", ValuesPath: ""}
+	NginxInternalCanonicalName: {Chart: "2.15.0", ValuesPath: ""},
 	// FluentbitDefaultVersion defines the default version for the Helm chart
-	FluentbitDefaultVersion = &HelmUtilityVersion{Chart: "0.15.8", ValuesPath: ""}
+	FluentbitCanonicalName: {Chart: "0.15.8", ValuesPath: ""},
 	// TeleportDefaultVersion defines the default version for the Helm chart
-	TeleportDefaultVersion = &HelmUtilityVersion{Chart: "0.3.0", ValuesPath: ""}
+	TeleportCanonicalName: {Chart: "0.3.0", ValuesPath: ""},
 	// PgbouncerDefaultVersion defines the default version for the Helm chart
-	PgbouncerDefaultVersion = &HelmUtilityVersion{Chart: "1.1.0", ValuesPath: ""}
+	PgbouncerCanonicalName: {Chart: "1.1.0", ValuesPath: ""},
 	// StackroxDefaultVersion defines the default version for the Helm chart
-	StackroxDefaultVersion = &HelmUtilityVersion{Chart: "62.0.0", ValuesPath: ""}
+	StackroxCanonicalName: {Chart: "62.0.0", ValuesPath: ""},
 	// KubecostDefaultVersion defines the default version for the Helm chart
-	KubecostDefaultVersion = &HelmUtilityVersion{Chart: "1.83.1", ValuesPath: ""}
-)
+	KubecostCanonicalName: {Chart: "1.83.1", ValuesPath: ""},
+}
 
 // SetUtilityDefaults is used to set Utility default version and values.
 func SetUtilityDefaults(url string) {
-	if PrometheusOperatorDefaultVersion.ValuesPath == "" {
-		PrometheusOperatorDefaultVersion.ValuesPath = fmt.Sprintf("%s/api/v4/projects/33/repository/files/dev%%2Fprometheus_operator_values.yaml?ref=master", url)
+	if DefaultUtilityVersions[PrometheusOperatorCanonicalName].ValuesPath == "" {
```

Comment:
Maybe we could refactor this to a map also? Something like `DefaultUtilityURLPath`?

Reply:
I'm going to try to clean this up a bit but it's a little bit of a mess, regardless, because we parameterize the base URL of the gitlab repo (that's the `url` argument here), but we hard-code the project number (`33`) so it's actually not possible to use the parameter to point to a different Gitlab instance, anyway.. unless you happen to make the project holding the repository project number 33 on the new instance, too.

I don't think this is how the original implementation worked but I think the code has been iterated upon to this state and the fact that we're hard-coding a configuration value (`33`, the project number) seems to have been overlooked until now. I'm going to spike on cleaning this up a bit.

Reply:
OK have a look at fa2b153 and let me know if you think I should spent more time making it possible to actually pass in the various parameters. I think that fa2b153 is an incremental improvement over the original code here, in any case, and it should make it easier for us to make these various parameters able to be passed in at runtime as necessary in the future, without altering the current behavior too much.